ga/client/bitrateadaptation.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2016 Chun-Ying Huang
+ *
+ * This file is part of GamingAnywhere (GA).
+ *
+ * GA is free software; you can redistribute it and/or modify it
+ * under the terms of the 3-clause BSD License as published by the
+ * Free Software Foundation: http://directory.fsf.org/wiki/License:BSD_3Clause
+ *
+ * GA is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * You should have received a copy of the 3-clause BSD License along with GA;
+ * if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <stdlib.h>
+#ifndef WIN32
+#include <unistd.h>
+#endif
+
+#include "ga-common.h"
+#include "controller.h"
+#include "rttserver.h"
+#include "bitrateadaptation.h"
+
+static struct bbr_btlbw_record_s bbr_btlbw[BBR_BTLBW_MAX];
+static unsigned int latest_throughput; // Not thread safe.
+static unsigned int bbr_btlbw_start = 0;
+static unsigned int bbr_btlbw_head = 0;
+static unsigned int last_pkt_timestamp = 0;
```

Comment:
Sorry, I made a mistake before. These were actually fine living inside the function.